co-simulation/bridge/carla_v0.9_xmlrpc_server.py

```diff
@@ -64,6 +68,21 @@ def __init__(self, host: str = 'localhost', port: int = 8090,
 
         self.lock = threading.RLock()
 
+        # Flag to track if first actor has been spawned (for automatic spectator switching)
+        self.first_actor_spawned = False
+
+        # Default extent_x for vehicle center calculation
+        self.default_extent_x = 2.0
+
+        # External-to-CARLA coordinate transform settings (SUMO/MOSAIC frame → CARLA frame)
+        # - input_frame: 'sumo' applies BridgeHelper-like conversion (y inversion, yaw - 90 deg, offset)
+        # - net_offset_xy: offset from SUMO net (x, y) applied before handedness flip
+        self.input_frame: str = 'sumo'
+        
+        # Try to read net offset from Town04.net.xml, fallback to hardcoded value
+      
+        self.net_offset_xy: Tuple[float, float] = (503.02, 423.76)
```

Comment:
I am not sure we want to have default offset here, for example, if we want to integrate other simulation in the future, this default will cause the problem since this is specific for SUMO